### PR TITLE
Bug 1936894: Reconcile the plan when a hook change is made

### PR DESCRIPTION
**Pull request by jmontleon**

### CODE REVIEW

File: pkg/controller/migplan/predicate.go [hunk 1]
```diff
@@ -85,6 +85,19 @@ func (r PlanPredicate) mapRefs(plan *migapi.MigPlan) {
 			Name:      ref.Name,
 		})
 	}
+
+	// hooks
+	for _, hook := range plan.Spec.Hooks {
```
Comment (djwhatle) [inline pkg/controller/migplan/predicate.go:90]: Needs a matching section in the `unmapRefs` function to handle the case where reference to a hook is removed: the MigPlan should no longer reconcile in response to MigHook change.

Comment (jmontleon) [reply pkg/controller/migplan/predicate.go:90]: Thank you. I'll fix it.

Comment (jmontleon) [reply pkg/controller/migplan/predicate.go:90]: Should be fixed, I believe.

**Review by djwhatle — APPROVED** [review]
VISACK

**Pull request merged**